lib/data_hygiene/title_fetcher.rb

```diff
@@ -0,0 +1,74 @@
+module DataHygiene
+  class TitleFetcher
+    # Fetch all topics from GovDelivery and update the subscriber lists in our
+    # database with the title for each if we don't have a title already. Log
+    # when we have a title which doesn't match GovDelivery's.
+
+    attr_reader :client, :logger, :stats
+
+    def initialize(client: Services.gov_delivery, logger: Logger.new(STDOUT))
+      @client = client
+      @logger ||= logger
+      @stats = Hash.new {|h, key| h[key] = [] }
+    end
+
+    def run
+      all_topics = fetch_all_topics
+
+      count = subscriber_lists.count
+      logger.info "Updating titles for #{count} subscriber lists"
+
+      subscriber_lists.each do |subscriber_list|
+        update_title(subscriber_list, all_topics[subscriber_list.gov_delivery_id])
+      end
+
+      logger.info ''
+      logger.info "Done:"
+      stats.each do |k, v|
+        logger.info "  #{k}: #{v.size}"
```

Comment:
Hm, I think I was going to use them in some tests to make sure that multiple lists were being counted in the right buckets, but looks like I didn't actually do that. The tests with single lists and on the logging for `#run` cover that enough, so I'll make it just keep count instead.